plugins/field-colour/src/field_colour.ts

```diff
@@ -137,32 +137,130 @@ export class FieldColour extends Blockly.Field<string> {
     this.size_ = new Blockly.utils.Size(
         constants.FIELD_COLOUR_DEFAULT_WIDTH,
         constants.FIELD_COLOUR_DEFAULT_HEIGHT);
-    if (!constants.FIELD_COLOUR_FULL_BLOCK) {
-      this.createBorderRect_();
-      this.getBorderRect().style['fillOpacity'] = '1';
-    } else if (this.sourceBlock_ instanceof Blockly.BlockSvg) {
-      this.clickTarget_ = this.sourceBlock_.getSvgRoot();
+    this.createBorderRect_();
+    this.getBorderRect().style['fillOpacity'] = '1';
+    this.getBorderRect().setAttribute('stroke', '#fff');
+    if (this.isFullBlockField()) {
+      this.clickTarget_ = (this.sourceBlock_ as Blockly.BlockSvg).getSvgRoot();
     }
   }
 
+  /**
+   * Defines whether this field should take up the full block or not.
+   * @returns True if this field should take up the full block. False otherwise.
+   */
+  protected isFullBlockField(): boolean {
+    const block = this.getSourceBlock();
+    if (!block) throw new Blockly.UnattachedFieldError();
+
+    const constants = this.getConstants();
+    return this.blockIsSimpleReporter() && !!constants?.FIELD_COLOUR_FULL_BLOCK;
```

Comment:
I'm not 100% sure about hte order of precedence here, and think it might be clearer to write:
```suggestion
    return this.blockIsSimpleReporter() && Boolean(constants?.FIELD_COLOUR_FULL_BLOCK);
```
(Assuming that that is indeed equivalent.)